cmd/prox/main.go

```diff
@@ -92,37 +94,39 @@ func environment(path string) (prox.Environment, error) {
 	return env, err
 }
 
-func processes(env prox.Environment, procFileFlag string) ([]prox.Process, error) {
-	var (
-		f     *os.File
-		err   error
-		parse = prox.ParseProxFile
-	)
-
-	if procFileFlag != "" {
-		// user has specified a path
-		logger.Debug("Reading processes from file specified via --procfile", zap.String("path", procFileFlag))
-		f, err = os.Open(procFileFlag)
-		if filepath.Base(procFileFlag) == "Procfile" {
-			parse = prox.ParseProcFile
-		}
-	} else {
-		// default to "Proxfile"
-		f, err = os.Open("Proxfile")
-		if os.IsNotExist(err) {
-			// no "Proxfile" but maybe we can open a "Procfile"
-			logger.Debug("Reading processes from Procfile")
-			parse = prox.ParseProcFile
-			f, err = os.Open("Procfile")
+func processes(env prox.Environment, path string) ([]prox.Process, error) {
+	var parser func(io.Reader, prox.Environment) ([]prox.Process, error)
+
+	switch {
+	case path == "" && fileExists("Proxfile"):
+		logger.Debug("Reading processes from Proxfile")
+		path = "Proxfile"
+		parser = prox.ParseProxFile
+	case path == "" && fileExists("Procfile"):
+		logger.Debug("Reading processes from Procfile")
+		path = "Procfile"
+		parser = prox.ParseProcFile
+	case path != "":
+		logger.Debug("Reading processes from file specified via --procfile", zap.String("path", path))
+		if strings.HasPrefix(filepath.Base(path), "Procfile") {
```

Comment:
@latortuga FYI: I decided to use a simpler prefix match instead of a regular expression. Let me know if you see any issues with this approach